src/hooks/useTonConnectUI.ts

```diff
@@ -6,25 +6,26 @@ import { isServerSide } from '../utils/web';
 export function useTonConnectUI() {
     if (isServerSide()) {
         return {
-            tonConnectUI: null as unknown as TonConnectUI,
+            tonConnectUI: null as TonConnectUI | null,
             setOptions: () => {},
         };
     }
 
-    const globalPropertiesMap = isVue2
-        ? getCurrentInstance()?.proxy
-        // @ts-expect-error
-        : getCurrentInstance().appContext.app.config.globalProperties;
+    const instance = getCurrentInstance() as  any;
+    const globalPropertiesMap = isVue2 ? instance?.proxy: instance?.appContext?.app.config.globalProperties
 
-    const tonConnectUI = globalPropertiesMap.$tonConnectUI;
+    const tonConnectUI = globalPropertiesMap?.$tonConnectUI as TonConnectUI | undefined;
+
+    if (!tonConnectUI) {
+        throw new Error("TonConnect is not avaliable");
```

Comment:
Throwing an exception directly within the hooks function body will interrupt the entire setup process. Consider using an error handler or logging a console error/warning to provide a notification instead.

Reply:
That’s an excellent question! I’ve carefully reconsidered this issue and have come up with a potential solution:
```js
return {
            tonConnectUI: null,
            setOptions: () => console.warn("TonConnect is not available"),
            error: new Error("TonConnect is not available"),
        };
```

demo
```js
const {tonConnectUI, setOptions, error} = useTonConnectUI();

    if (error) {
      console.warn(error.message); // 输出警告信息
      return {}; // 或者在 UI 上显示提示
    }

```